sql/index/pilosa/driver.go

```diff
@@ -140,12 +142,50 @@ var (
 	errDeletePilosaIndex = errors.NewKind("error deleting pilosa index %s: %s")
 )
 
+type bitBatch struct {
+	size uint64
+	bits []pilosa.Bit
+	pos  uint64
+}
+
+func newBitBatch(size uint64) *bitBatch {
+	b := &bitBatch{size: size}
+	b.Clean()
+
+	return b
+}
+
+func (b *bitBatch) Clean() {
+	b.bits = make([]pilosa.Bit, 0, b.size)
+	b.pos = 0
+}
+
+func (b *bitBatch) Add(row, col uint64) {
+	b.bits = append(b.bits, pilosa.Bit{
+		RowID:    row,
+		ColumnID: col,
+	})
+}
+
+func (b *bitBatch) NextRecord() (pilosa.Record, error) {
+	if b.pos >= uint64(len(b.bits)) {
+		return nil, io.EOF
+	}
+
+	b.pos++
+	return b.bits[b.pos-1], nil
+}
+
+func (b *bitBatch) Send(frame *pilosa.Frame, client *pilosa.Client) error {
```

Comment:
maybe this function we can add to the driver and just call `sendFrame` and pass `b` instead of `client` - wdyt

Reply:
It was done like this as I tried to put most of the new functionality in `bitBatch`. It's the one that implements the interface that is used by `ImportFrame` call.

But yes, it also makes sense using client only in `Driver`. We can change this if you believe it is better. We also have to send `frame` to the call.